src/eth/TransactionLifeCycle.js

```diff
@@ -62,6 +64,11 @@ class TransactionLifeCycle {
    * @returns {boolean|null}
    */
   isFinalized() {
+    if (
+      this._blockNumberWhenMined + this._confirmedBlockCount <=
+      this._web3Service.blockNumber()
+    )
+      this.setFinalized();
```

Comment:
Making sure we set the state to execute the callbacks.

Reply:
This is looking good too. Just one more case to handle: when the user does call `confirm`, and they pass a non-default argument for the number of blocks to wait. I think in this case the `confirm` method should actually change the value of `this._confirmedBlockCount` to what was passed as an argument, so that `isFinalized` can use it here to do its check. Otherwise, calling `isFinalized` would result in a finalized callback running too early.

Reply:
I just noticed that there's a code smell here: we're adding code to the superclass that depends upon instance variables set by a subclass. It would be better add an `isFinalized` to `TransactionObject` that has this logic and then calls `return super.isFinalized()`.

Reply:
Oh, and then we don't need to add the web3 service to the constructor :)

Reply:
Yep, I see what you're saying, I'll implement it that way. 👍